src/api/index.md

```diff
@@ -0,0 +1,277 @@
+returns a new reference to a function assigned those properties. This is then
+partially applied to return a new function that takes properties. Each time the
+function is called a new reference is created to enable the ability for new
+entities to be created from a primitive version of itself carrying along its own
+properties and at the beginning starting with none.
+
+### Quick intro
+
+`blockType()`, `block()`, `control()`, and `controlType()`, all serve as an independent
+tool for building larger pieces. Each are identical other than their name. They
+all work from the same fundamental pattern. Let's look at `blockType`, and
+`controlType`. These would serve as a way to create factories for `block()` and
+control(). A good idea might be to have a `create()` method on a type, which would
+create the actual block, or control. Then each block or control would have a
+`render()` method which would create an actual representation of that block or
+control. `render()` could be used to return React components, DOM, vDOM, HTML,
+TinyMCE, whatever is chosen. Let's look at creating a `blockType`.
+
+### Creating a textBlockType
+
+```js
+// textBlock would be the block
+import textBlock from '../blocks/textBlock'
+import blockType from './blockType'
+
+const textBlockTypeProperties = {
+	type: 'wp-text',
+	title: 'Text',
+	controls: [
+		[
+			controlType( {
+				type: 'left-align',
+				title: 'Align Left',
+				displayArea: 'block'
+			} ),
+			controlType( {
+				type: 'center-align',
+				title: 'Center Align',
+				displayArea: 'block'
+			} ),
+			controlType( {
+				type: 'right-align',
+				title: 'Align Right',
+				displayArea: 'block'
+			} ),
+			controlType( {
+				type: 'bold',
+				title: 'Make Bold',
+				displayArea: 'inline'
+			} ),
+			controlType( {
+				type: 'italics',
+				title: 'Italicize',
+				displayArea: 'inline'
+			} ),
+			controlType( {
+				type: 'strikethrough',
+				title: 'Strikethrough',
+				displayArea: 'inline'
+			} ),
+			controlType( {
+				type: 'underline',
+				title: 'Underline',
+				displayArea: 'inline'
+			} )
+		]
+	]
+	create: ( blockType = this ) => textBlock( blockType )
+}
+
+const textBlockType = blockType( textBlockTypeProperties )
+
+// Create a text block.
+const aTextBlock = textBlockType.create()
+
+// Then grab the DOM node it creates.
+const myTextDOMNode = aTextBlock.render( 'Blocks are neat.' )
+```
+
+### Creating a textBlock
+
+```js
+// Example text block.
+const textBlock = blockType => {
+    return block ( {
+        /**
+         * State and props would be objects that would be passed in via data held somewhere. i.e. the UI
+         *
+         * The render method is defined by you, so you can make it work the way you want it to.
+         */
+        render: ( state = {}, props = {} ) => {
+            // Do some stuff relating to the blockType for this block.
+
+            // Render something. Could be DOM, vDOM, React, TinyMCE, JSON, XML, HTML, whatever you want.
```

Comment:
> Would we expect developers to implement all/each of these, or is this just outlining all of the options we would want to support? Should we have a preference toward one single recommendation for rendering?

This could be whatever you want it to be, but probably it should be a set interface like this taking `state` and `props`.  Potentially it should just be a state object.  I did state props here to match React.